### ISSUE TITLE
Repository test suite does not run

### ISSUE THREAD

**Issue by ctrl-f5** [Codeception/Codeception#6047]
#### What are you trying to achieve?

I want to make a PR. to do this I want to clone the repo and run the tests.

#### What do you get instead?

a failing test suite:

as per documentation in https://github.com/Codeception/Codeception/blob/4.1/tests/README.md:

    git clone git@github.com:Codeception/Codeception.git 
    cd Codeception/tests 
    docker-compose build
    docker-compose run --rm codecept run cli

Does not run without failures:

    ERRORS!
    Tests: 208, Assertions: 659, Errors: 11, Failures: 22, Skipped: 3.

As a side note:

> https://github.com/Codeception/Codeception/blob/4.1/CONTRIBUTING.md

states

> Breaking features and major improvements should be sent into master. 

But there is no master branch in the repo.

Full output:

```
Time: 00:28.123, Memory: 14.00 MB

There were 11 errors:

---------
1) GeneratePageObjectCest: Generate global page object in different path
 Test  tests/cli/GeneratePageObjectCest.php:generateGlobalPageObjectInDifferentPath
                                                                                                                                          
  [InjectionException] Failed to inject dependencies in instance of 'GeneratePageObjectCest'. Class CliGuy\GeneratorSteps does not exist  
                                                                                                                                          
#1  /repo/app.php:46
#2  /repo/app.php:47
#3  /repo/codecept:7

---------
2) SnapshotCest: Generate suite snapshot
 Test  tests/cli/SnapshotCest.php:generateSuiteSnapshot
                                                                                                                                
  [InjectionException] Failed to inject dependencies in instance of 'SnapshotCest'. Class CliGuy\GeneratorSteps does not exist  
                                                                                                                                
#1  /repo/app.php:46
#2  /repo/app.php:47
#3  /repo/codecept:7

---------
3) GeneratePageObjectCest: Generate global page object
 Test  tests/cli/GeneratePageObjectCest.php:generateGlobalPageObject
                                                                                                                                          
  [InjectionException] Failed to inject dependencies in instance of 'GeneratePageObjectCest'. Class CliGuy\GeneratorSteps does not exist  
                                                                                                                                          
#1  /repo/app.php:46
#2  /repo/app.php:47
#3  /repo/codecept:7

---------
4) GeneratePageObjectCest: Generate suite page object
 Test  tests/cli/GeneratePageObjectCest.php:generateSuitePageObject
                                                                                                                                          
  [InjectionException] Failed to inject dependencies in instance of 'GeneratePageObjectCest'. Class CliGuy\GeneratorSteps does not exist  
                                                                                                                                          
#1  /repo/app.php:46
#2  /repo/app.php:47
#3  /repo/codecept:7

---------
5) GenerateGroupCept: Generate new group
 Test  tests/cli/GenerateGroupCept.php
                                                   
  [Error] Class 'CliGuy\GeneratorSteps' not found  
                                                   
#1  /repo/tests/cli/GenerateGroupCept.php:2
#2  /repo/app.php:46
#3  /repo/app.php:47
#4  /repo/codecept:7

---------
6) GenerateStepObjectCept: Generate step object
 Test  tests/cli/GenerateStepObjectCept.php
                                                   
  [Error] Class 'CliGuy\GeneratorSteps' not found  
                                                   
#1  /repo/tests/cli/GenerateStepObjectCept.php:2
#2  /repo/app.php:46
#3  /repo/app.php:47
#4  /repo/codecept:7

---------
7) GenerateCestCept: Generate sample Cest
 Test  tests/cli/GenerateCestCept.php
                                                   
  [Error] Class 'CliGuy\GeneratorSteps' not found  
                                                   
#1  /repo/tests/cli/GenerateCestCept.php:2
#2  /repo/app.php:46
#3  /repo/app.php:47
#4  /repo/codecept:7

---------
8) SnapshotCest: Generate global snapshot
 Test  tests/cli/SnapshotCest.php:generateGlobalSnapshot
                                                                                                                                
  [InjectionException] Failed to inject dependencies in instance of 'SnapshotCest'. Class CliGuy\GeneratorSteps does not exist  
                                                                                                                                
#1  /repo/app.php:46
#2  /repo/app.php:47
#3  /repo/codecept:7

---------
9) SnapshotCest: Generate global snapshot in different path
 Test  tests/cli/SnapshotCest.php:generateGlobalSnapshotInDifferentPath
                                                                                                                                
  [InjectionException] Failed to inject dependencies in instance of 'SnapshotCest'. Class CliGuy\GeneratorSteps does not exist  
                                                                                                                                
#1  /repo/app.php:46
#2  /repo/app.php:47
#3  /repo/codecept:7

---------
10) GenerateHelperCept: Generate helper
 Test  tests/cli/GenerateHelperCept.php
                                                   
  [Error] Class 'CliGuy\GeneratorSteps' not found  
                                                   
#1  /repo/tests/cli/GenerateHelperCept.php:2
#2  /repo/app.php:46
#3  /repo/app.php:47
#4  /repo/codecept:7

---------
11) GenerateTestCept: Generate sample Test
 Test  tests/cli/GenerateTestCept.php
                                                   
  [Error] Class 'CliGuy\GeneratorSteps' not found  
                                                   
#1  /repo/tests/cli/GenerateTestCept.php:2
#2  /repo/app.php:46
#3  /repo/app.php:47
#4  /repo/codecept:7

--

There were 22 failures:

---------
1) ConfigParamsCest: Check dot env params passed
 Test  tests/cli/ConfigParamsCest.php:checkDotEnvParamsPassed
 Step  Execute command "run --debug -g pass2"
 Fail  Result code was 1.


In ParamsLoader.php line 51:

  Failed loading params from params.env
  `vlucas/phpdotenv` library is required to parse .env files.
  Please install it via composer: composer require vlucas/phpdotenv



Scenario Steps:

 2. $I->executeCommand("run -c codeception_dotenv.yml") at tests/cli/ConfigParamsCest.php:15
 1. $I->amInPath("tests/data/params") at tests/cli/ConfigParamsCest.php:14


---------
2) RunCest: Run tests with steps
 Test  tests/cli/RunCest.php:runTestsWithSteps
 Step  Execute command "run -c codeception_self.yml"
 Fail  Result code was 2.

Codeception PHP Testing Framework v4.1.12
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.
Running with seed:


Scenario Steps:

 2. $I->executeCommand("run scenario SuccessCept --steps") at tests/cli/RunCest.php:319
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
3) ConfigParamsCest: Check complex dot env params passed
 Test  tests/cli/ConfigParamsCest.php:checkComplexDotEnvParamsPassed
 Step  See result code is 0
 Fail  Result code was 1.


In ParamsLoader.php line 51:

  Failed loading params from params.env.example
  `vlucas/phpdotenv` library is required to parse .env files.
  Please install it via composer: composer require vlucas/phpdotenv



Scenario Steps:

 2. $I->executeCommand("run -c codeception_dotenv2.yml") at tests/cli/ConfigParamsCest.php:22
 1. $I->amInPath("tests/data/params") at tests/cli/ConfigParamsCest.php:21


---------
4) RunCest: Run warning tests
 Test  tests/cli/RunCest.php:runWarningTests
 Step  See in shell output "tests => tests"
 Fail  Failed asserting that 'Codeception PHP Testing Framework v4.1.12\n
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.\n
Running with seed:\n
\n
\n
Unit Tests (1) -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------' contains "There was 1 warning".

Scenario Steps:

 3. $I->seeInShellOutput("There was 1 warning") at tests/cli/RunCest.php:565
 2. $I->executeCommand("run unit WarningTest.php",false) at tests/cli/RunCest.php:564
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
5) RunCest: Reports correct failed step
 Test  tests/cli/RunCest.php:reportsCorrectFailedStep
 Step  Execute command "run dummy"
 Fail  Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run scenario File.feature -v") at tests/cli/RunCest.php:423
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
6) RunCest: Show same order of files on seed
 Test  tests/cli/RunCest.php:showSameOrderOfFilesOnSeed
 Step  Execute command "run powers PowerUpCest"
 Fail  Failed asserting that '' contains "Seed".

Scenario Steps:

 3. $I->seeInShellOutput("Seed") at tests/cli/RunCest.php:594
 2. $I->executeCommand("run unit -o "settings: shuffle: true"",false) at tests/cli/RunCest.php:593
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
7) GroupExtensionCept: See that my group extension works
 Test  tests/cli/GroupExtensionCept.php
 Step  Execute command "run tests/dummy/FileExistsCept.php"
 Fail  Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run skipped -g notorun -c codeception_grouped.yml") at tests/cli/GroupExtensionCept.php:5
 1. $I->amInPath("tests/data/sandbox") at tests/cli/GroupExtensionCept.php:4


---------
8) OrderCest: Check after before class in tests
 Test  tests/cli/OrderCest.php:checkAfterBeforeClassInTests
 Step  Execute command "run tests/dummy/FileExistsCept.php -c codeception_extended.yml"
 Fail  Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run order BeforeAfterClassTest.php") at tests/cli/OrderCest.php:97
 1. $I->amInPath("tests/data/sandbox") at tests/cli/OrderCest.php:96


---------
9) RunCest: Check xml reports
 Test  tests/cli/RunCest.php:runXmlReport
 Step  Execute command "run tests/dummy/AnotherCest.php:optimistic -c codeception_extended.yml"
 Fail  Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run dummy --xml") at tests/cli/RunCest.php:75
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
10) RetryCest: Check the test fails even after retries
 Test  tests/cli/RetryCest.php:checkTheTestFailsEvenAfterRetries
 Step  Execute command "run unit FailingTest.php -c codeception_extended.yml --no-exit"
 Fail  Failed asserting that 'Codeception PHP Testing Framework v4.1.12\n
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.\n
Running with seed:\n
\n
[Groups] fail1\n
\n
Retry Tests (1) ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n
Modules: \Helper\Retry\n
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\n
FailCest: Fail num\n
Signature: FailCest:failNum\n
Test: tests/retry/FailCest.php:failNum\n
Scenario --\n
 I fail at 3\n
  Retrying #1 in 200ms' contains "Retrying #2".

Scenario Steps:

 4. $I->seeInShellOutput("Retrying #2") at tests/cli/RetryCest.php:28
 3. $I->seeResultCodeIsNot(0) at tests/cli/RetryCest.php:27
 2. $I->executeFailCommand("run --debug -g fail1") at tests/cli/RetryCest.php:26
 1. $I->amInPath("tests/data/retries") at tests/cli/RetryCest.php:7


---------
11) RunCest: Run tests with dependency injections
 Test  tests/cli/RunCest.php:runTestsWithDependencyInjections
 Step  Execute command "run dummy -g countevents -c codeception_grouped.yml --no-colors"
 Fail  Result code was 2.

Codeception PHP Testing Framework v4.1.12
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.
Running with seed:


Math Tests (6) -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
✔ MathCest: Test addition (0.00s)
✔ MathCest: Test subtraction (0.00s)
✔ MathCest: Test square (0.00s)
✔ MathCest: Test trigonometry (0.00s)

Scenario Steps:

 2. $I->executeCommand("run math") at tests/cli/RunCest.php:289
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
12) RetryCest: Check that retry happens on specific interval
 Test  tests/cli/RetryCest.php:checkThatRetryHappensOnSpecificInterval
Result code was 2.

Codeception PHP Testing Framework v4.1.12
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.
Running with seed:

[Groups] pass2

Retry Tests (1) ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Modules: \Helper\Retry
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FailCest: Pass time1
Signature: FailCest:passTime1
Test: tests/retry/FailCest.php:passTime1
Scenario --
 I fail for 0.6
  Retrying #1 in 200ms

Scenario Steps:

 2. $I->executeCommand("run --debug -g pass2") at tests/cli/RetryCest.php:19
 1. $I->amInPath("tests/data/retries") at tests/cli/RetryCest.php:7


---------
13) ConfigParamsCest: Check params passed in self
 Test  tests/cli/ConfigParamsCest.php:checkParamsPassedInSelf
Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run -c codeception_self.yml") at tests/cli/ConfigParamsCest.php:37
 1. $I->amInPath("tests/data/params") at tests/cli/ConfigParamsCest.php:36


---------
14) ConfigNoActorCest: Suites without actor generators
 Test  tests/cli/ConfigNoActorCest.php:suitesWithoutActorGenerators
result code is 0
Failed asserting that 0 matches expected 2.

Scenario Steps:

 7. $I->seeResultCodeIs(0) at tests/cli/ConfigNoActorCest.php:34
 6. $I->executeFailCommand("generate:test unit Some") at tests/cli/ConfigNoActorCest.php:33
 5. $I->seeResultCodeIsNot(0) at tests/cli/ConfigNoActorCest.php:32
 4. $I->executeFailCommand("generate:cest unit Some") at tests/cli/ConfigNoActorCest.php:31
 3. $I->seeResultCodeIsNot(0) at tests/cli/ConfigNoActorCest.php:30
 2. $I->executeFailCommand("generate:cept unit Some") at tests/cli/ConfigNoActorCest.php:29


---------
15) ConfigValidateCest: Prints valid config
 Test  tests/cli/ConfigValidateCest.php:printsValidConfig
Failed asserting that '' contains "tests => tests".

Scenario Steps:

 4. $I->seeInShellOutput("tests => tests") at tests/cli/ConfigValidateCest.php:14
 3. $I->dontSeeInShellOutput("ConfigurationException") at tests/cli/ConfigValidateCest.php:13
 2. $I->executeCommand("config:validate --no-ansi",false) at tests/cli/ConfigValidateCest.php:12
 1. $I->amInPath("tests/data/sandbox") at tests/cli/ConfigValidateCest.php:7


---------
16) RunCest: Skip group of cest
 Test  tests/cli/RunCest.php:skipGroupOfCest
Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run dummy") at tests/cli/RunCest.php:201
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
17) RunCest: Run test with custom setup method
 Test  tests/cli/RunCest.php:runTestWithCustomSetupMethod
Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run powers PowerUpCest") at tests/cli/RunCest.php:548
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
18) RunCest: Execute one test
 Test  tests/cli/RunCest.php:runOneFile
Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run tests/dummy/FileExistsCept.php") at tests/cli/RunCest.php:15
 1. $I->amInPath("tests/data/sandbox") at tests/cli/RunCest.php:9


---------
19) ExtensionsCest: Use alternative formatter delivered through extensions
 Test  tests/cli/ExtensionsCest.php:useAlternativeFormatter
Result code was 2.



Scenario Steps:

 2. $I->executeCommand("run tests/dummy/FileExistsCept.php -c codeception_extended.yml") at tests/cli/ExtensionsCest.php:10
 1. $I->amInPath("tests/data/sandbox") at tests/cli/ExtensionsCest.php:9


---------
20) ExtensionsCest: Check if extensions receive correct options
 Test  tests/cli/ExtensionsCest.php:checkIfExtensionsReceiveCorrectOptions
Result code was 1.

Codeception PHP Testing Framework v4.1.12
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.
Running with seed:


Scenario Steps:

 2. $I->executeCommand("run tests/dummy/AnotherCest.php:optimistic -c codeception_extended.yml") at tests/cli/ExtensionsCest.php:58
 1. $I->amInPath("tests/data/sandbox") at tests/cli/ExtensionsCest.php:57


---------
21) ExtensionsCest: Re run failed tests
 Test  tests/cli/ExtensionsCest.php:reRunFailedTests
Result code was 1.

Codeception PHP Testing Framework v4.1.12
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.
Running with seed:


Scenario Steps:

 2. $I->executeCommand("run unit FailingTest.php -c codeception_extended.yml --no-exit") at tests/cli/ExtensionsCest.php:38
 1. $I->amInPath("tests/data/sandbox") at tests/cli/ExtensionsCest.php:36


---------
22) GroupEventsCept: See that my group events fire only once
 Test  tests/cli/GroupEventsCept.php
Result code was 1.

Codeception PHP Testing Framework v4.1.12
Powered by PHPUnit 9.5.0 by Sebastian Bergmann and contributors.
Running with seed:

[Groups] countevents

Scenario Steps:

 2. $I->executeCommand("run dummy -g countevents -c codeception_grouped.yml --no-colors") at tests/cli/GroupEventsCept.php:5
 1. $I->amInPath("tests/data/claypit") at tests/cli/GroupEventsCept.php:4


ERRORS!
Tests: 208, Assertions: 659, Errors: 11, Failures: 22, Skipped: 3.
```

**Comment by ctrl-f5** [comment]
Latest branch and tag builds also seem to be failing, so I guess there are some issues with the test suite...

- 4.1 branch: https://travis-ci.org/github/Codeception/Codeception/builds/747808477
- 4.1.13 tag: https://travis-ci.org/github/Codeception/Codeception/builds/743877170

**Comment by Naktibalda** [comment]
Build failures most likely are caused by PHPUnit 9.5,
either try to fix them or add <=9.5 constraint to composer.json

If nobody fixes them sooner, I will fix them myself, but I can't promise to do it before Christmas.

**Comment by Naktibalda** [comment]
Only WarningTest error was caused by PHPUnit 9.5
